11.0 imp product_pricelist_direct_print

ernestotejeda opened the pull request:
It must be rebased when PR #370  is merged

* Print price list for future date
* Print price list for N last ordered products
* Send report by email to each partner selected

Cc @Tecnativa

ernestotejeda committed:
product_pricelist_direct_print: extend module

Print price list for future date
Print price list for N last ordered products
Send report by email to each partner selected

sergio-teruel commented on product_pricelist_direct_print/wizards/product_pricelist_print.py:
```diff
@@ -0,0 +1,162 @@
+            if len(active_ids) == 1:
+                res['partner_id'] = active_ids[0]
+                partner = self.env['res.partner'].browse(active_ids[0])
+                res['pricelist_id'] = partner.property_product_pricelist.id
```
```suggestion
                res['pricelist_id'] = partner.property_product_pricelist.id
        elif self.env.context.get('active_model') == 'product.pricelist.item':
            active_ids = self.env.context.get('active_ids', [])
            items = self.env['product.pricelist.item'].browse(active_ids)
            if items[:1].applied_on == '0_product_variant':
                res['show_variants'] = True
                res['product_ids'] = [(6, 0, items.mapped('product_id').ids)]
            else:
                res['product_tmpl_ids'] = [
                    (6, 0, items.mapped('product_tmpl_id').ids)]               
```

sergio-teruel replied:
Can you add this code? is for print from pricelist items..
Thanks

sergio-teruel commented on product_pricelist_direct_print/wizards/product_pricelist_print.py:
```diff
@@ -0,0 +1,162 @@
+    @api.multi
+    def force_pricelist_send(self):
+        if self.env.context.get('active_model') != 'res.partner':
+            return False
```
Remove this condition to allow send email from any model

sergio-teruel commented on product_pricelist_direct_print/wizards/product_pricelist_print.py:
```diff
@@ -0,0 +1,162 @@
+            'default_model': 'product.pricelist.print',
+            'default_template_id': template_id,
+            'active_ids': self.ids,
+            'partner_to': self.partner_id,
```
Remove partner_to key

sergio-teruel commented on product_pricelist_direct_print/wizards/product_pricelist_print_view.xml:
```diff
@@ -0,0 +1,98 @@
+                            <field name="name"/>
+                        </tree>
+                    </field>
+                </page>
```
```suggestion
                </page>
                <page string="Customers">
                    <field name="partner_ids"/>
                    <field name="partner_count" invisible="1"/>
                </page>                
```

sergio-teruel replied:
Allow select partners.

sergio-teruel commented on product_pricelist_direct_print/wizards/product_pricelist_print_view.xml:
```diff
@@ -0,0 +1,98 @@
+    res_model="product.pricelist.print"
+    groups="product.group_sale_pricelist"
+    view_mode="form" target="new" view_type="form"/>
+
```
```suggestion

<act_window id="action_item_pricelist_print"
    key2="client_print_multi"
    name="Price List"
    res_model="product.pricelist.print"
    src_model="product.pricelist.item"
    groups="product.group_sale_pricelist"
    view_mode="form" target="new" view_type="form"/>

```

sergio-teruel replied:
Print from pricelists items

sergio-teruel commented:
ping @ernestotejeda

ernestotejeda commented:
Thanks @sergio-teruel, I have made almost all the changes you propose except the ones related with pricelist items because I don't know where I can find their corresponding tree view to be able to test those changes.
When you talk about selecting several partners, to be consistent with the behavior of the module should be sent to each one an email with the report corresponding to the associated price list, and therefore I have added some other things like, Once at least one partner is selected, the pricelists in the wizard is not visible, and some details about the visibility of the fields I changed too.

pedrobaeza commented:
You can rebase this PR now that I have merged the other one.

chienandalu commented:
Well there's no such pricelist item list view :sweat_smile:

chienandalu reviewed (commented):
I can't make multiple partner send work on runbot:

![image](https://user-images.githubusercontent.com/5040182/49144046-a8a70a80-f2fc-11e8-8f6a-16e347d7df2f.png)

ernestotejeda commented:
@chienandalu is true that in *Settings -> Technical -> Email -> Messages* the corresponding messages have the field *Recipients* empty, but emails are sent correctly.
If you test with incorrect addresses you will see in *Settings -> Technical -> Email -> Emails* that the emails have the field *To (Partners) * correctly, but if you test with correct addresses, emails are deleted but those ones are sent correctly .
Apparently this is the behavior of mass mailing. When module *mass_mailing* is installed and you send a newsletter you will see the same behavior.

pedrobaeza commented:
> Thanks @sergio-teruel, I have made almost all the changes you propose except the ones related with pricelist items because I don't know where I can find their corresponding tree view to be able to test those changes.

@ernestotejeda this is because in standard there's no such view... unless you go to the model in *Settings > Technical > Database structure > Model*, find that model, and add menu item. Please do it and add the suggested code.

chienandalu commented:
@ernestotejeda I'm trying to PR to your repo this commit improving the pricelist items print but I can't find your fork: https://github.com/Tecnativa/product-attribute/commit/7bbc7a43915c38eafee863fba94d343188948eae

- Sets the pricelist in the wizard if all the items belong to the same one.
- Lets to print items applied on different rules altogether.

ping @sergio-teruel

pedrobaeza reviewed (approved):
I don't like too much to have you fields `partner_id` and `partner_ids` for the same thing, as you can check if `len(partner_ids) == 1`, but for not blocking this more time and not doing diff bigger, let's merge it this way.
